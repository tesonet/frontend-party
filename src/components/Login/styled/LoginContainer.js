import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import {
  Container,
  Row,
  Col,
} from 'reactstrap';

import BackgroundImage from '../../../assets/background.jpg';

const StyledContainer = styled.div`
  background-color: ${props => props.theme.bgColorDark};
  background-image: url(${BackgroundImage});
  background-size: cover;
  display: flex;
  flex-direction: column;
  height: 100%;
  justify-content: center;
`;

// There is an invisible alert box of 5.5rem height above the logo,
// so we need to center out the form view using this method.
const StyledRow = styled(Row)`
  margin-top: -5.5rem;
`;

const LoginContainer = props => (
  <StyledContainer>
    <Container>
      <StyledRow>
        <Col
          xs={{ size: 12 }}
          sm={{ size: 8, offset: 2 }}
          md={{ size: 6, offset: 3 }}
          lg={{ size: 4, offset: 4 }}
        >
          {props.children}
        </Col>
      </StyledRow>
    </Container>
  </StyledContainer>
);

LoginContainer.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
};

export default LoginContainer;
