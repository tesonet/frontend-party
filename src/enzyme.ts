import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import fetchMock from "jest-fetch-mock";

declare let global: { fetch: {} };

global.fetch = fetchMock;
configure({ adapter: new Adapter() });
