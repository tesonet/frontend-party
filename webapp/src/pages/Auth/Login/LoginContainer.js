import React, { Component } from 'react';
import { connect } from 'react-redux';

import LoginView from './LoginView';
import { doLogin } from '../authActions';
import publicRoute from '../../../hoc/publicRoute';
import api from '../../../utils/api';

export const MSG_ERROR_USERNAME_EMPTY = 'Username cannot be empty.';
export const MSG_ERROR_PASSWORD_EMPTY = 'Password cannot be empty.';
export const MSG_ERROR_UNAUTHORIZED = 'Your username or password is incorrect.';
export const MSG_ERROR_GLOBAL = 'Something went wrong. Try again later.';

export const ROUTE_PATH = '/login';

export class LoginContainer extends Component {
  state = {
    hasSubmitted: false,
    isBusy: false,
    globalError: null,
    errors: null,
    username: '',
    password: '',
  };

  componentDidMount() {
    this.setState({ errors: this.validate() });
  }

  validate() {
    const errors = {};
    const { username, password } = this.state;

    if (username === '') {
      errors.username = MSG_ERROR_USERNAME_EMPTY;
    }

    if (password === '') {
      errors.password = MSG_ERROR_PASSWORD_EMPTY;
    }

    return Object.keys(errors).length === 0 ? null : errors;
  }

  onChange = ({ target }) => {
    this.setState(
      { [target.name]: target.value },
      () => this.setState({ errors: this.validate() })
    );
  };

  onSubmit = (e) => {
    e.preventDefault();

    const errors = this.validate();
    this.setState({ hasSubmitted: true, errors });

    if (errors !== null) return;

    this.setState({ isBusy: true });

    return api
      .tokens
      .post(this.state.username, this.state.password)
      .then(tokens => {
        this.setState({ isBusy: false });
        api.setToken(tokens.token);
        this.props.doLogin();
      })
      .catch((e) => {
        let errorMsg;

        if (e.response && e.response.status === 401) {
          errorMsg = MSG_ERROR_UNAUTHORIZED;
        } else {
          errorMsg = MSG_ERROR_GLOBAL;
        }

        this.setState({ isBusy: false, globalError: errorMsg });
      });
  };

  render() {
    return (
      <LoginView
        showErrors={ this.state.hasSubmitted }
        globalError={ this.state.globalError }
        errors={ this.state.errors }
        isDisabled={ !! (this.state.isBusy || this.state.errors) }
        username={ this.state.username }
        password={ this.state.password }
        onSubmit={ this.onSubmit }
        onChange={ this.onChange }
      />
    );
  }
}

export default publicRoute(connect(null, { doLogin })(LoginContainer));
