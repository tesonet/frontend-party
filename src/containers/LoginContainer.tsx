import React from 'react'
import BaseContainer from './BaseContainer'
import { useSelector } from 'react-redux'
import Login from 'components/Login/Login'

const LoginContainer: React.FC = () => {
  const loading = useSelector(s => s.auth.loading)
  return <BaseContainer content={<Login />} loading={loading} />
}
export default LoginContainer
