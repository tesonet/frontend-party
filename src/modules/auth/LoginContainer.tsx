import * as background from '@assets/background.jpg';
import { BodyBackground } from '@components/BodyBackground';
import { styled } from '@components/theme';
import * as React from 'react';

import { Login } from './Login';

const LoginPageContainer = styled.div`
  height: 100%;
  margin: 0 auto;
  max-width: 320px;
  display: flex;
  align-content: center;
  justify-content: center;
`;

const LoginContainer: React.FunctionComponent = () => {
  return (
    <LoginPageContainer>
      <BodyBackground background={background} />
      <Login />
    </LoginPageContainer>
  );
};

export { LoginContainer };
