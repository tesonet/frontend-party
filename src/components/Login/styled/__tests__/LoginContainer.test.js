import React from 'react';
import { shallow } from 'enzyme';

import LoginContainer from '../LoginContainer';

describe('LoginContainer component', () => {
  it('should render the <div /> element.', () => {
    const renderedComponent = shallow(
      <LoginContainer>Test</LoginContainer>
    ).dive();

    expect(renderedComponent.type()).toEqual('div');
    expect(renderedComponent).toMatchSnapshot();
  });
});
