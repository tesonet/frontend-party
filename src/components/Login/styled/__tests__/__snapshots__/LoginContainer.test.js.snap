// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginContainer component should render the <div /> element. 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <styled.div>
    <Container
        tag="div"
    >
        <Styled(Row)>
            <Col
                lg={
                    Object {
                        "offset": 4,
                        "size": 4,
                      }
                }
                md={
                    Object {
                        "offset": 3,
                        "size": 6,
                      }
                }
                sm={
                    Object {
                        "offset": 2,
                        "size": 8,
                      }
                }
                tag="div"
                widths={
                    Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ]
                }
                xs={
                    Object {
                        "size": 12,
                      }
                }
            >
                Test
            </Col>
        </Styled(Row)>
    </Container>
</styled.div>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Container
        tag="div"
>
        <Styled(Row)>
                <Col
                        lg={
                                Object {
                                        "offset": 4,
                                        "size": 4,
                                      }
                        }
                        md={
                                Object {
                                        "offset": 3,
                                        "size": 6,
                                      }
                        }
                        sm={
                                Object {
                                        "offset": 2,
                                        "size": 8,
                                      }
                        }
                        tag="div"
                        widths={
                                Array [
                                        "xs",
                                        "sm",
                                        "md",
                                        "lg",
                                        "xl",
                                      ]
                        }
                        xs={
                                Object {
                                        "size": 12,
                                      }
                        }
                >
                        Test
                </Col>
        </Styled(Row)>
</Container>,
      "className": "sc-bdVaJa itKpME",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Styled(Row)>
          <Col
                    lg={
                              Object {
                                        "offset": 4,
                                        "size": 4,
                                      }
                    }
                    md={
                              Object {
                                        "offset": 3,
                                        "size": 6,
                                      }
                    }
                    sm={
                              Object {
                                        "offset": 2,
                                        "size": 8,
                                      }
                    }
                    tag="div"
                    widths={
                              Array [
                                        "xs",
                                        "sm",
                                        "md",
                                        "lg",
                                        "xl",
                                      ]
                    }
                    xs={
                              Object {
                                        "size": 12,
                                      }
                    }
          >
                    Test
          </Col>
</Styled(Row)>,
        "tag": "div",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Col
            lg={
                        Object {
                                    "offset": 4,
                                    "size": 4,
                                  }
            }
            md={
                        Object {
                                    "offset": 3,
                                    "size": 6,
                                  }
            }
            sm={
                        Object {
                                    "offset": 2,
                                    "size": 8,
                                  }
            }
            tag="div"
            widths={
                        Array [
                                    "xs",
                                    "sm",
                                    "md",
                                    "lg",
                                    "xl",
                                  ]
            }
            xs={
                        Object {
                                    "size": 12,
                                  }
            }
>
            Test
</Col>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Test",
            "lg": Object {
              "offset": 4,
              "size": 4,
            },
            "md": Object {
              "offset": 3,
              "size": 6,
            },
            "sm": Object {
              "offset": 2,
              "size": 8,
            },
            "tag": "div",
            "widths": Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ],
            "xs": Object {
              "size": 12,
            },
          },
          "ref": null,
          "rendered": "Test",
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Container
          tag="div"
>
          <Styled(Row)>
                    <Col
                              lg={
                                        Object {
                                                  "offset": 4,
                                                  "size": 4,
                                                }
                              }
                              md={
                                        Object {
                                                  "offset": 3,
                                                  "size": 6,
                                                }
                              }
                              sm={
                                        Object {
                                                  "offset": 2,
                                                  "size": 8,
                                                }
                              }
                              tag="div"
                              widths={
                                        Array [
                                                  "xs",
                                                  "sm",
                                                  "md",
                                                  "lg",
                                                  "xl",
                                                ]
                              }
                              xs={
                                        Object {
                                                  "size": 12,
                                                }
                              }
                    >
                              Test
                    </Col>
          </Styled(Row)>
</Container>,
        "className": "sc-bdVaJa itKpME",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Styled(Row)>
            <Col
                        lg={
                                    Object {
                                                "offset": 4,
                                                "size": 4,
                                              }
                        }
                        md={
                                    Object {
                                                "offset": 3,
                                                "size": 6,
                                              }
                        }
                        sm={
                                    Object {
                                                "offset": 2,
                                                "size": 8,
                                              }
                        }
                        tag="div"
                        widths={
                                    Array [
                                                "xs",
                                                "sm",
                                                "md",
                                                "lg",
                                                "xl",
                                              ]
                        }
                        xs={
                                    Object {
                                                "size": 12,
                                              }
                        }
            >
                        Test
            </Col>
</Styled(Row)>,
          "tag": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Col
              lg={
                            Object {
                                          "offset": 4,
                                          "size": 4,
                                        }
              }
              md={
                            Object {
                                          "offset": 3,
                                          "size": 6,
                                        }
              }
              sm={
                            Object {
                                          "offset": 2,
                                          "size": 8,
                                        }
              }
              tag="div"
              widths={
                            Array [
                                          "xs",
                                          "sm",
                                          "md",
                                          "lg",
                                          "xl",
                                        ]
              }
              xs={
                            Object {
                                          "size": 12,
                                        }
              }
>
              Test
</Col>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Test",
              "lg": Object {
                "offset": 4,
                "size": 4,
              },
              "md": Object {
                "offset": 3,
                "size": 6,
              },
              "sm": Object {
                "offset": 2,
                "size": 8,
              },
              "tag": "div",
              "widths": Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ],
              "xs": Object {
                "size": 12,
              },
            },
            "ref": null,
            "rendered": "Test",
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
